### ISSUE TITLE
aks create fails with 'Namespace' object has no attribute 'detach_acr'

### ISSUE THREAD

**Issue by geNAZt** [Azure/azure-cli-extensions#1147]
## Describe the bug
I updated my azure CLI from .68 to .77 to enable managed identities on AKS. After that i tried to vcreate a new AKS and it failed with the error below

**Command Name**
`az aks create
Extension Name: aks-preview. Version: 0.4.24.`

**Errors:**
az aks create: error: 'Namespace' object has no attribute 'detach_acr'

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az aks create --name {} --resource-group {} --kubernetes-version {} --network-plugin {} --subscription {} --tags {} {} {} --location {} --attach-acr {} --enable-vmss --node-vm-size {} --enable-cluster-autoscaler --min-count {} --max-count {} --windows-admin-password {} --windows-admin-username {} --enable-managed-identity`

## Expected Behavior
AKS cluster gets created

## Environment Summary
```
Darwin-19.0.0-x86_64-i386-64bit
Python 3.7.5
Shell: bash / zsh

azure-cli 2.0.77

Extensions:
aks-preview 0.4.24

```

**Comment by adrianwyatt** [comment]
Repros even without --enable-managed-identity

### Command
`az aks create --resource-group myrg --name myaks--generate-ssh-keys --attach-acr myacr`

### Environment Summary
```
azure-cli                         2.0.77 *

command-modules-nspkg              2.0.3
core                              2.0.77 *
nspkg                              3.0.4
telemetry                          1.0.4

Extensions:
aks-preview                       0.4.24
application-insights               0.1.1
azure-devops                       0.3.0
```

**Comment by palma21** [comment]
@sauryadas for ACR property on the extension.

**Comment by sauryadas** [comment]
@norshtein is working on this

**Comment by elangelo** [comment]
any updates on this? I don't even know how to work around the issue?

**Comment by palma21** [comment]
the attach command (GA) is not on the extension CLI where the msi command is (PREVIEW), we will pull it to the extension CLI.

If you want you can create the cluster with MSI with the extension and them manually give permissions to ACR. 

When the rollout finishes we will publish it here.

**Comment by palma21** [comment]
In the meantime you can create the cluster normally with MSI and give permissions to ACR for the Managed Identity manually afterwards. What the attach-acr is doing is just creating an acr-pull role assignment to that identity. 

We will publish when this is rolled out on the extension as well.

**Comment by dhilgarth** [comment]
Can you please communicate the workaround as a series of commands instead of prose?

So, what commands do we have to execute instead of `az aks create --resource-group myrg --name myaks--generate-ssh-keys --attach-acr myacr`?

**Comment by gfilicetti** [comment]
@dhilgarth You basically can't use --attach-acr when creating the cluster.

So run your az aks create command without that option, and then run:

`az aks update -n myCluster -g myRG --attach-acr myregistryname`

**Comment by sauryadas** [comment]
@feiskyer Can you please help prioritize this?

@dhilgarth  if you can remove the extension it should work

**Comment by feiskyer** [comment]
The option is broken by some new changes, @norshtein is working on a fix. 

/assign @norshtein

**Comment by alecor191** [comment]
I followed instructions from @gfilicetti (create cluster, then `az aks update`), but unfortunately I still get the following error when trying to deploy containers:

```
Failed to pull image "...": rpc error: code = Unknown desc = Error response from daemon: Get https://<myacr>.azurecr.io/v2/...: unauthorized: authentication required
```

I created the cluster with `--enable-managed-identity`. The `az aks update` command also completed successfully.

Anything I'm missing?

**Comment by feiskyer** [comment]
@alecor191 the issue is being fixed by https://github.com/Azure/azure-cli-extensions/pull/1217

**Comment by navba-MSFT** [comment]
@geNAZt Apologies for the late reply. This issue seems to be fixed as per #1217.
Could you please use the [most recent version](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli) of the CLI and try to run [this](https://docs.microsoft.com/en-us/azure/aks/cluster-container-registry-integration?tabs=azure-cli#create-a-new-aks-cluster-with-acr-integration) Azure CLI command and let us know if you still face this issue ? Awaiting your reply.

**Comment by navba-MSFT** [comment]
@geNAZt This is a follow-up to check if you had a chance to look at my above comment. This issue is open for quite sometime. Please let us know if you need any further assistance on this.
